torch/distributed/distributed_c10d.py

```diff
@@ -3388,7 +3418,7 @@ def _create_process_group_wrapper(
     wrapped_pg = _ProcessGroupWrapper(wrapped_pg, helper_pg)
     return wrapped_pg
 
-def new_group(ranks=None, timeout=default_pg_timeout, backend=None, pg_options=None):
+def new_group(ranks=None, timeout=default_pg_timeout, backend=None, pg_options=None, pg_tag=None):
```

Comment:
new_group is pretty widely used and in our public API. Is there a way to supporting functional collectives without changing this API until we are certain about the `pg_tag` design. If this change is inevitable, please create an RFC in github issue to explain the motivation and design for pg_tag and properly document it.

Reply:
Good point, I moved it to an internal function and move it back to new_group once functional collectives take shape.